tfhe/src/core_crypto/fft_impl/f128/crypto/bootstrap.rs

```diff
@@ -0,0 +1,439 @@
+            let mut ct0 = lut;
+
+            for (lwe_mask_element, bootstrap_key_ggsw) in
+                izip!(lwe_mask.iter(), this.into_ggsw_iter())
+            {
+                if *lwe_mask_element != Scalar::ZERO {
+                    let stack = stack.rb_mut();
+                    // We copy ct_0 to ct_1
+                    let (mut ct1, stack) =
+                        stack.collect_aligned(CACHELINE_ALIGN, ct0.as_ref().iter().copied());
+                    let mut ct1 =
+                        GlweCiphertextMutView::from_container(&mut *ct1, ct0.polynomial_size());
+
+                    // We rotate ct_1 by performing ct_1 <- ct_1 * X^{a_hat}
+                    for mut poly in ct1.as_mut_polynomial_list().iter_mut() {
+                        polynomial_wrapping_monic_monomial_mul_assign(
+                            &mut poly,
+                            MonomialDegree(pbs_modulus_switch(
+                                *lwe_mask_element,
+                                lut_poly_size,
+                                ModulusSwitchOffset(0),
+                                LutCountLog(0),
+                            )),
+                        );
+                    }
+
+                    // ct1 is re-created each loop it can be moved, ct0 is already a view, but
+                    // as_mut_view is required to keep borrow rules consistent
+                    cmux(&mut ct0, &mut ct1, &bootstrap_key_ggsw, fft, stack);
+                }
+            }
+        }
+        implementation(self.as_view(), lut.as_mut_view(), lwe.as_view(), fft, stack)
+    }
+
+    pub fn bootstrap<Scalar, ContLweOut, ContLweIn, ContAcc>(
+        &self,
+        lwe_out: &mut LweCiphertext<ContLweOut>,
+        lwe_in: &LweCiphertext<ContLweIn>,
+        accumulator: &GlweCiphertext<ContAcc>,
+        fft: Fft128View<'_>,
+        stack: PodStack<'_>,
+    ) where
+        // CastInto required for PBS modulus switch which returns a usize
+        Scalar: UnsignedTorus + CastInto<usize>,
+        ContLweOut: ContainerMut<Element = Scalar>,
+        ContLweIn: Container<Element = Scalar>,
+        ContAcc: Container<Element = Scalar>,
+    {
+        fn implementation<Scalar: UnsignedTorus + CastInto<usize>>(
+            this: Fourier128LweBootstrapKey<&[f64]>,
+            mut lwe_out: LweCiphertext<&mut [Scalar]>,
+            lwe_in: LweCiphertext<&[Scalar]>,
+            accumulator: GlweCiphertext<&[Scalar]>,
+            fft: Fft128View<'_>,
+            stack: PodStack<'_>,
+        ) {
+            if TypeId::of::<Scalar>() == TypeId::of::<u128>() {
+                let mut lwe_out: LweCiphertext<&mut [u128]> = unsafe { transmute(lwe_out) };
+                let lwe_in: LweCiphertext<&[u128]> = unsafe { transmute(lwe_in) };
+                let accumulator: GlweCiphertext<&[u128]> = unsafe { transmute(accumulator) };
+
+                return this.bootstrap_u128(&mut lwe_out, &lwe_in, &accumulator, fft, stack);
+            }
```

Comment:
maybe add a comment to indicate we are dispatch to the u128 specific implem here :) I did not catch it at first !